src/modules/watch/service.js

```diff
@@ -35,12 +50,14 @@ module.exports = class WatchService {
       payload: { watchID: insertedId }
     })
 
-    return ops[0] // this should be aggregatedWatch, need to be fixed later
+    return ops[0]
   }
 
   async getByID(id) {
     const _id = new ObjectID(id)
-    return this.watchCollection.findOne({ _id }) // this should be aggregatedWatch, need to be fixed later
+    const watch = await this.watchCollection.findOne({ _id })
+    const aggregatedWatch = await this.aggregate(watch)
+    return aggregatedWatch
```

Comment:
```suggestion
    return this.aggregate(watch)
```
Các chỗ khác cũng viết thế này nhé, cho tiện, đỡ 1 lần await. Thực tế nó cũng sẽ tăng chút hiệu năng đấy.